From f075f9c6cbf11c6484330d24b8a49fbbdb465d66 Mon Sep 17 00:00:00 2001
From: Eric Huss <eric@huss.org>
Date: Sun, 4 Nov 2018 10:38:51 -0800
Subject: [PATCH] Fix can_run_doc_tests order depends on hash.

The deps are sorted, but the name is the same so the order depends on the metadata hash.
Fix by sorting by the actual name, too.
---
 src/cargo/core/compiler/context/mod.rs | 2 ++
 tests/testsuite/rename_deps.rs         | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/cargo/core/compiler/context/mod.rs b/src/cargo/core/compiler/context/mod.rs
index 225e6ebd53..334a3876aa 100644
--- a/src/cargo/core/compiler/context/mod.rs
+++ b/src/cargo/core/compiler/context/mod.rs
@@ -234,6 +234,8 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
                         }
                     }
                 }
+                // Help with tests to get a stable order with renamed deps.
+                doctest_deps.sort();
                 self.compilation.to_doc_test.push(compilation::Doctest {
                     package: unit.pkg.clone(),
                     target: unit.target.clone(),
diff --git a/tests/testsuite/rename_deps.rs b/tests/testsuite/rename_deps.rs
index c444739b0f..209987b89c 100644
--- a/tests/testsuite/rename_deps.rs
+++ b/tests/testsuite/rename_deps.rs
@@ -334,8 +334,8 @@ fn can_run_doc_tests() {
 [DOCTEST] foo
 [RUNNING] `rustdoc --test [CWD]/src/lib.rs \
         [..] \
-        --extern baz=[CWD]/target/debug/deps/libbar-[..].rlib \
         --extern bar=[CWD]/target/debug/deps/libbar-[..].rlib \
+        --extern baz=[CWD]/target/debug/deps/libbar-[..].rlib \
         [..]`
 ",
         ).run();
